File: binary_search/binary_search.js
// Difficulty: O(log(n))
function binarySearch(list, value) {
    let low = 0
    let high = list.length - 1

    while (low <= high) {
        const mid = Math.floor((low + high) / 2)

        if (list[mid] === value) {
            return mid
        } else if (list[mid] > value) {
            high = mid - 1
        } else {
            low = mid + 1
        }
    }

    return -1
}

module.exports = binarySearch

File: binary_search/binary_search.go
package binary_search

// Difficulty: O(log(n))
func BinarySearchStr(list []string, value string) int {
	// We skip it this time
	// return sort.SearchStrings(list, value)

	low := 0
	high := len(list) - 1

	for low <= high {
		mid := (low + high) / 2

		if list[mid] == value {
			return mid
		} else if list[mid] > value {
			high = mid - 1
		} else {
			low = mid + 1
		}
	}

	return -1
}
